generic/tests/edk2_stability_test.py

```diff
@@ -0,0 +1,46 @@
+import re
+
+from avocado.core import exceptions
+from virttest import error_context
+from virttest import utils_misc
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu edk2 stability test:
+    1) Try to log into a guest
+    2) Check serial log information
+    3) Cycle the above process
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    timeout = params.get_numeric("login_timeout", 40)
+    check_messgae = params.get("check_messgae")
+
+    def boot_check(info):
+        """
+        boot info check
+        """
+        return re.search(info, get_serial_console_output(), re.S)
+
+    def get_serial_console_output():
+        """
+        get the output of serail console
+        """
+        output = vm.serial_console.get_output()
+        return output
+
+    for i in range(int(params.get("reboot_count", 1))):
+        params["start_vm"] = "yes"
+        env_process.preprocess(test, params, env)
+        vm = env.get_vm(params.get("main_vm"))
```

Comment:
Based on Zhenchao's previous suggestion, try best to avoid modifying a cartesian param at runtime. We can use it in the following way.
cfg: 
start_vm = no
py:
vm_name = params["main_vm"]
env_process.preprocess_vm(test, params, env, vm_name)
vm = env.get_vm(vm_name)
\# start_vm = no, not run vm.create in preprocess_vm, so need to add it
vm.create(params=params)
vm.verify_alive()

Just FYI, all look good to me.

Reply:
As `virt_test_type = qemu libvirt`, so qemu(vm.create) == libvirt(vm.start). Keeping this is okay.

BTW, @zhenyzha, can you provide test result on using libvirt?

Reply:
Thanks for reminding, this case will only run on tp-qemu, corrected.